script.js

```diff
@@ -1,4 +1,339 @@
+  // Loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+      } else if (cardName == 12) {
+        cardName = "Queen";
+      } else if (cardName == 13) {
+        cardName = "King";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+var createNewDeck = function () {
+  var newDeck = createDeck();
+  var shuffledDeck = shuffleCards(newDeck);
+  return shuffledDeck;
+};
+
+var checkPlayerForBlackJack = function () {
+  var playerTotalValue = 0;
+  for (var i = 0; i < playerCards.length; i++) {
+    var cardValue = playerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      if (playerTotalValue + 11 <= 21) {
+        cardValue = 11;
+      } else {
+        cardValue = 1;
+      }
+    }
+    playerTotalValue += Number(cardValue);
+  }
+  if (playerTotalValue == 21) {
+    return true;
+  } else {
+    return false;
+  }
+};
+var checkDealerForBlackJack = function () {
```

Comment:
Consider having just one `checkForBlackJack(player)` function and you can pass in either player or dealer into the parameter